pages/index.tsx

```diff
@@ -0,0 +1,584 @@
+          from_id: systemMessageLinkId,
+          to_id: conversationLinkId,
+          in: {
+            data: {
+              type_id: containTypeLinkId,
+              from_id: deep.linkId,
+            },
+          },
+        });
+
+        const { data: [{ id: messageLinkId }] } = await deep.insert({
+          type_id: messageTypeLinkId,
+          string: { data: { value: transcribedTextLinkId.value?.value } },
+          in: {
+            data: {
+              type_id: containTypeLinkId,
+              from_id: deep.linkId,
+            }
+          },
+        });
+
+        const { data: [{ id: replyToMessageLinkId }] } = await deep.insert({
+          type_id: replyTypeLinkId,
+          from_id: messageLinkId,
+          to_id: conversationLinkId,
+          in: {
+            data: {
+              type_id: containTypeLinkId,
+              from_id: deep.linkId,
+            },
+          },
+        });
+      };
+      console.log("flakeed8");
+      setSounds([]);
+    };
+
+    if (sounds.length > 0) useRecords();
+  }, [sounds, isRecording]);
+
+  useEffect(() => {
+    if (!isRecording) return;
+
+    let loop = true;
+    const startRecordingCycle = async (duration) => {
+      for (; isRecording && loop;) {
+        await startAudioRec(deep);
+        const startTime = new Date().toLocaleDateString();
+        await delay(duration);
+        const record = await stopAudioRec(deep);
+        const endTime = new Date().toLocaleDateString();
+        console.log({ record });
+        setSounds([{ record, startTime, endTime }]);
+      }
+    };
+
+    startRecordingCycle(5000);
+    return function stopCycle() {
+      loop = false;
+    };
+  }, [isRecording]);
+
+  const ScreenChat = ({ replyToMessageLinkId }) => {
+    const [messages, setMessages] = useState([]);
+    const [messagesCount, setMessagesCount] = useState(0);
+    let chatGptLinkId;
+    async () => { chatGptLinkId = await deep.id('@deep-foundation/chatgpt', 'ChatGPT') }
+
+    useEffect(() => {
+      (async () => {
+        const messagingTreeId = await deep.id("@deep-foundation/messaging", "MessagingTree");
+        const messageTypeLinkId = await deep.id("@deep-foundation/messaging", "Message");
+        const authorTypeLinkId = await deep.id("@deep-foundation/messaging", "Author");
+
+        const result = await deep.select({
+          tree_id: { _eq: messagingTreeId },
+          link: { type_id: { _eq: messageTypeLinkId } },
+          root_id: { _eq: replyMessageLinkId },
+          // @ts-ignore
+          self: { _eq: true },
+        }, {
+          table: 'tree',
+          variables: { order_by: { depth: "asc" } },
+          returning: `
+            id
+            depth
+            root_id
+            parent_id
+            link_id
+            link {
+              id
+              from_id
+              type_id
+              to_id
+              value
+              author: out (where: { type_id: { _eq: ${authorTypeLinkId} } }) { 
```

Comment:
Have not we added author to the tree? I remember that we have added it to the tree and we were able to just select links up(or down, I do not remember) to dialog. Is not it?